src/rpc/claimtrie.cpp

```diff
@@ -741,8 +742,7 @@ UniValue getnameproof(const UniValue& params, bool fHelp)
     uint256 blockHash;
     if (params.size() == 2)
     {
-        std::string strBlockHash = params[1].get_str();
-        blockHash = uint256S(strBlockHash);
+        uint256 hash = ParseHashV(params[1], "blockhash (optional parameter 2)");
```

Comment:
This needs to set blockHash, otherwise its just setting a variable that never gets used